Allow unprivileged mode when running as splunk user

mikedickey opened the pull request:
This PR enables running in unprivileged mode when run as the splunk user account. The splunk user is now part of the ansible group and able to run the ansible playbooks itself at startup, but unlike ansible, it has no sudo capabilities at startup time.

Note that you currently must explicitly set SPLUNK_HOME_OWNERSHIP_ENFORCEMENT to false when running as the splunk user, otherwise it will fail due to lack of permissions.

Note that there are limitations, beyond just being unable to "correct" permission for volume mounts. Any features requiring elevated permissions, such as install JDK or other packages, will not work when running as the splunk user.

The defaults are left unchanged. By default, it will still run as the ansible user and the behavior should be the same as before.

Commit:
This PR enables running in unpriviliged mode when run as the splunk
user account. The splunk user is now part of the ansible group and
able to run the ansible playbooks itself at startup, but unlike ansible,
it has no sudo capabilities at startup time.

Note that you currently must explicitly set SPLUNK_HOME_OWNERSHIP_ENFORCEMENT
to false when running as the splunk user, otherwise it will fail due to
lack of permissions.

Note that there are limitations, beyond just being unable to "correct"
permission for volume mounts. Any features requiring elevated permissions,
such as install JDK or other packages, will not work when running as the
splunk user.

The defaults are left unchanged. By default, it will still run as the
ansible user and the behavior should be the same as before.

mikedickey commented:
Also note I made some indenting fixes in entrypoint.sh so that it's (I think) all using tabs now

bb03 commented on splunk/common-files/entrypoint.sh:
```diff
@@ -89,58 +95,58 @@ configure_multisite() {
 help() {
 	cat << EOF
-  ____        _             _      __
+	____        _             _      __
```
I think the tab to spaces messed with our ascii art here.  If we're going all in on tabs, can you fix the spacing here.

Commit:
More whitespace changes in splunk/splunk entrypoint.sh file to fix
ascii art formatting and finish indenting consistency.

Merged in same changes for running as splunk user to universal
forwarder Dockerfile and entrypoint.sh.

Updated whitespace in universal forwarder entrypoint.sh to also use
tables for consistency.

nwang92 commented:
Overall, this follows what @lephino and I came up with last Friday so I don't have any real issues with it. I can add some tests for any upgrade functionality so we have that in our pipeline in the future.

nwang92 commented on splunk/common-files/entrypoint.sh:
```diff
@@ -35,6 +35,9 @@ trap teardown SIGINT SIGTERM
 prep_ansible() {
 	cd ${SPLUNK_ANSIBLE_HOME}
+	if [ `whoami` == "${SPLUNK_USER}" ]; then
+		sed -i -e "s,^become\\s*=.*,become = false," ansible.cfg
```
One small thing here that I don't quite like is that our `ansible.cfg` as shown in the repo is not actually representative of the ansible.cfg during execution under a different user. But I suppose this is a necessary evil at the moment so as long as it's documented, it should be alright.

mikedickey replied:
Yeah, I was hoping that `become_user` when you are that user is a noop, but unfortuantely that is not the case. Without explicitly disabling it here, ansible bombs out at startup with complaints that it cannot use sudo without a password.

Commit:
Added SECURITY.md documentation

sayeedc commented:
I’ve encountered issues running Splunk SE and UF on Openshift in unprivileged mode. When will this PR be put into an image release on either the Redhat container catalog or Docker Hub?

mikedickey commented:
> I’ve encountered issues running Splunk SE and UF on Openshift in unprivileged mode. When will this PR be put into an image release on either the Redhat container catalog or Docker Hub?

Hi Sayeed, this should be included in the `edge` containers (which are built nightly), and will be included with the next release (7.3.2?). For OpenShift, I'd recommend trying specifically `splunk/splunk:edge-redhat`.

sayeedc commented:
> > I’ve encountered issues running Splunk SE and UF on Openshift in unprivileged mode. When will this PR be put into an image release on either the Redhat container catalog or Docker Hub?
> 
> Hi Sayeed, this should be included in the `edge` containers (which are built nightly), and will be included with the next release (7.3.2?). For OpenShift, I'd recommend trying specifically `splunk/splunk:edge-redhat`.

Hi Mike. Thanks for the info. I'll test out the edge image. Is there an edge-redhat image for the the universal forwarder?
